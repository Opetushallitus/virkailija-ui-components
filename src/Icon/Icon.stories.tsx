import * as React from 'react';
import { storiesOf } from '@storybook/react';

import Icon from './index';

const stories = storiesOf('Components/Icon', module);

stories.add('Default', () => (
  <>
    <Icon type="remove" color="danger.main" fontSize="1.25rem" />
    <Icon type="add" color="success.main" ml={2} />
  </>
));
