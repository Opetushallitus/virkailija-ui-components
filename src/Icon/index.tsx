import * as React from 'react';
import styled from 'styled-components';

import {
  color,
  ColorProps,
  space,
  SpaceProps,
  typography,
  TypographyProps,
} from '../system';

type SystemProps = ColorProps & SpaceProps & TypographyProps;

const IconBase = styled.i<SystemProps>`
  font-size: 1.5rem;
  ${color};
  ${space};
  ${typography};
`;

type MaterialIconTheme =
  | 'outlined'
  | 'two-tone'
  | 'round'
  | 'sharp'
  | undefined;

export type IconProps = React.ComponentProps<typeof IconBase> & {
  type: string;
  variant: MaterialIconTheme;
};

const Icon = React.forwardRef<HTMLSpanElement, IconProps>(
  ({ type, variant, className = '', ...props }, ref) => {
    const iconClassName = `material-icons${
      variant ? '-' + variant : ''
    } ${className}`;

    return (
      <IconBase ref={ref} className={iconClassName} {...props}>
        {type}
      </IconBase>
    );
  },
);

export default Icon;
