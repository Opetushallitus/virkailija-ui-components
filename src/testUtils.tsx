import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { createTheme } from './createTheme';

const theme = createTheme();

const Providers = ({ children }: { children: React.ReactNode }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const dummyDiv = document.createElement('div');
dummyDiv.setAttribute('data-nothing', 'true');

// Return rootElement from render() for easier snapshot testing
const customRender = (ui: React.ReactElement, options: any = {}) => {
  const result = render(ui, { wrapper: Providers, ...options });
  return {
    ...result,
    rootElement: result.container.firstChild || dummyDiv,
  };
};

export * from '@testing-library/react';

export { customRender as render };
