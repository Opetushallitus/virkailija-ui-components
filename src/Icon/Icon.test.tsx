import React from 'react';
import { render } from '../testUtils';
import Icon from './index';

it('Renders an icon', () => {
  const { rootElement } = render(<Icon type="flag" />);
  expect(rootElement).toMatchSnapshot();
});

it('Renders an outlined icon', () => {
  const { rootElement } = render(<Icon type="flag" variant="outlined" />);
  expect(rootElement).toMatchSnapshot();
});
