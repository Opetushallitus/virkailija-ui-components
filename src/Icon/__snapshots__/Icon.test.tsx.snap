// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders an icon 1`] = `
.c0 {
  font-size: 1.5rem;
}

<i
  class="c0 material-icons "
>
  flag
</i>
`;

exports[`Renders an outlined icon 1`] = `
.c0 {
  font-size: 1.5rem;
}

<i
  class="c0 material-icons-outlined "
>
  flag
</i>
`;
